core/src/main/java/org/frankframework/pipes/IsUserInRolePipe.java

```diff
@@ -58,36 +61,49 @@ public void configure() throws ConfigurationException {
 		if (StringUtils.isNotEmpty(getNotInRoleForwardName())) {
 			notInRoleForward = findForward(getNotInRoleForwardName());
 			if (notInRoleForward==null) {
-				throw new ConfigurationException("notInRoleForwardName ["+getNotInRoleForwardName()+"] not found");
+				throw new ConfigurationException("notInRoleForwardName not found");
 			}
 		}
 	}
 
-	protected void assertUserIsInRole(PipeLineSession session, String role) throws SecurityException {
-		if (!session.isUserInRole(role)) {
-			throw new SecurityException("user is not in role ["+role+"]");
+	protected String inWhichRoleIsUser(PipeLineSession session, List<String> roles) {
+		ISecurityHandler securityHandler = session.getSecurityHandler();
+		return roles.stream()
+				.filter(securityHandler::isUserInRole)
+				.findFirst().orElse(null);
+	}
+
+	protected List<String> getRolesToCheck(Message message) throws PipeRunException {
+		String roles = getRole();
+		if (StringUtils.isEmpty(roles)) {
+			try {
+				roles = message.asString();
+			} catch (IOException e) {
+				throw new PipeRunException(this, "cannot open stream", e);
+			}
 		}
+		return StringUtil.split(roles);
 	}
 
 	@Override
 	public PipeRunResult doPipe(Message message, PipeLineSession session) throws PipeRunException {
 		try {
-			if (StringUtils.isEmpty(getRole())) {
-				String inputString;
-				try {
-					inputString = message.asString();
-				} catch (IOException e) {
-					throw new PipeRunException(this, "cannot open stream", e);
-				}
-				if (StringUtils.isEmpty(inputString)) {
-					throw new PipeRunException(this, "role cannot be empty");
-				}
-				assertUserIsInRole(session, inputString);
-			} else {
-				assertUserIsInRole(session, getRole());
+			List<String> rolesToCheck = getRolesToCheck(message);
+			if (rolesToCheck.isEmpty()) {
+				throw new PipeRunException(this, "role cannot be empty");
+			}
+
+			String userRole = inWhichRoleIsUser(session, rolesToCheck);
+			if (userRole == null) {
+				throw new SecurityException("user is not in role(s) [" + rolesToCheck + "]");
+			}
+
+			PipeForward relatedForward = findForward(userRole);
+			if (relatedForward != null) {
+				return new PipeRunResult(relatedForward, message);
 			}
 		} catch (SecurityException e) {
-			if (notInRoleForward!=null) {
+			if (notInRoleForward!=null) { // is this needed? configure() test for != null..?
```

Comment:
Good question, but in the configure I already see a null check being performed, on this value. So, feel free to remove the redundant check here.